frontend/public/components/operator-lifecycle-manager/create-operand.tsx

```diff
@@ -50,17 +55,70 @@ const fieldsFor = (providedAPI: CRDDescription) => _.get(providedAPI, 'specDescr
 })) as OperandField[];
 
 export const CreateOperandForm: React.FC<CreateOperandFormProps> = (props) => {
-  type FormValues = {[path: string]: any};
-  const [formValues, setFormValues] = React.useState<FormValues>({'metadata.name': 'example', 'metadata.labels': []});
-  const [error, setError] = React.useState();
-
   const fields: OperandField[] = !_.isEmpty(props.clusterServiceVersion)
     ? fieldsFor(providedAPIsFor(props.clusterServiceVersion).find(desc => referenceForProvidedAPI(desc) === referenceForProvidedAPI(props.providedAPI)))
     : [];
 
+  const defaultValueFor = (field: OperandField) => {
+    if (field.capabilities.includes(SpecCapability.podCount)) {
+      return 1;
+    }
+    if (field.capabilities.includes(SpecCapability.resourceRequirements)) {
+      return {limits: {cpu: '', memory: ''}, requests: {cpu: '', memory: ''}};
+    }
+    if (field.capabilities.includes(SpecCapability.password)) {
+      return '';
+    }
+    if (field.capabilities.some(c => c.startsWith(SpecCapability.k8sResourcePrefix))) {
+      return null;
+    }
+    if (field.capabilities.includes(SpecCapability.checkbox)) {
+      return false;
+    }
+    if (field.capabilities.includes(SpecCapability.booleanSwitch)) {
+      return false;
+    }
+    if (field.capabilities.includes(SpecCapability.imagePullPolicy)) {
+      return ImagePullPolicy.IfNotPresent;
+    }
+    if (field.capabilities.includes(SpecCapability.updateStrategy)) {
+      return 'Rolling';
+    }
+    if (field.capabilities.includes(SpecCapability.text)) {
+      return '';
+    }
+    if (field.capabilities.includes(SpecCapability.number)) {
+      return '';
+    }
+    if (field.capabilities.includes(SpecCapability.nodeAffinity)) {
+      return null;
+    }
+    if (field.capabilities.includes(SpecCapability.podAffinity) || field.capabilities.includes(SpecCapability.podAntiAffinity)) {
+      return null;
+    }
+    return null;
+  };
+
+  type FormValues = {[path: string]: any};
+
+  const defaultFormValues = fields.reduce((allFields, field) => ({...allFields, [field.path]: defaultValueFor(field)}), {} as FormValues);
+  const sampleFormValues = fields.reduce((allFields, field) => {
+    const sampleValue = _.get(props.sample, `spec.${field.path}`);
+    return sampleValue ? {...allFields, [field.path]: sampleValue} : allFields;
+  }, {} as FormValues);
+
+  const [formValues, setFormValues] = React.useState<FormValues>({'metadata.name': 'example', 'metadata.labels': [], ...defaultFormValues, ...sampleFormValues});
+  const [error, setError] = React.useState();
+
+  const updateFormValues = (values: FormValues) => (path: string, value: any) => {
+    return _.set(_.cloneDeep(values), path, value);
```

Comment:
We can't just use `_.set()` directly because it mutates the given object; this was causing the component _not_ to rerender because [React compares the previous and current state before deciding to rerender](https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update). So we have this little function instead.